app/models/ems_event/automate.rb

```diff
@@ -12,6 +12,19 @@ def refresh(*targets, sync)
       EmsRefresh.queue_refresh(refresh_targets, nil, sync)
     end
 
+    def refresh_new_target
+      ems = ext_management_system
+      if ems.supports_refresh_new_target?
+        event_data = {
+          :vm      => full_data[:vm],
+          :cluster => full_data[:cluster],
+          :message => message,
+          :ems_id  => ems_id
+        }
+        EmsRefresh.queue_refresh_new_target(event_data, ems)
+      end
```

Comment:
I am not sure about it. The goal for my changes is to get rid of full refresh but if we are going to use this mechanism for other providers as well I think that we could have it here.